src/vision.cpp

```diff
@@ -380,7 +394,8 @@ bool Vision::publish()
   // Update header information
   auto cur_cinfo = camera_info_manager_->getCameraInfo();
 
-  if (cur_cinfo.height != image_height_ || cur_cinfo.width != image_width_)
+  if (cur_cinfo.height != static_cast<unsigned int>(image_height_) ||
+      cur_cinfo.width != static_cast<unsigned int>(image_width_))
```

Comment:
I think you should go the other way. Cast the `cur_cinfo.height/width` to `int` https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es102-use-signed-types-for-arithmetic